test/acme/dns/fixture.go

```diff
@@ -96,7 +99,28 @@ func (f *fixture) setup(t *testing.T) func() {
 	f.clientset = cl
 
 	stopCh := make(chan struct{})
-	f.testSolver.Initialize(env.Config, stopCh)
+
+	var testSolver webhook.Solver
+	switch f.testSolverType {
+	case rfc2136.SolverName:
+		cl, err := kubernetes.NewForConfig(env.Config)
+		if err != nil {
+			t.Errorf("error initializing solver: %#+v", err)
+		}
+
+		// obtain a secret lister and start the informer factory to populate the
+		// secret cache
+		factory := informers.NewSharedInformerFactoryWithOptions(cl, time.Minute*5)
+		secretLister := factory.Core().V1().Secrets().Lister()
+		factory.Start(stopCh)
+		factory.WaitForCacheSync(stopCh)
+		testSolver = rfc2136.New(rfc2136.WithSecretsLister(secretLister))
+		f.testSolver = testSolver
+	default:
+		t.Errorf("unknown solver type: %s", f.testSolverType)
+	}
+
+	testSolver.Initialize(env.Config, stopCh)
```

Comment:
Predefined lister gets passed in instead of relying on the nil check in `Initialize` because this same solver gets reused in multiple revocations of this function (which shouldn't be how its ever used in real life, so that should be okay)